gordo_components/builder/build_model.py

```diff
@@ -121,15 +126,16 @@ def build(
             if self.cached_model_path:
                 model = serializer.load(self.cached_model_path)
                 metadata = serializer.load_metadata(self.cached_model_path)
-                metadata["metadata"]["user-defined"] = self.machine.metadata
+                metadata["metadata"][
+                    "user_defined"
+                ] = self.machine.metadata.user_defined
+                machine = Machine(**metadata)
 
             # Otherwise build and cache the model
             else:
-                model, metadata = self._build()
+                model, machine = self._build()
                 self.cached_model_path = self._save_model(
-                    model=model,
-                    metadata=metadata,
-                    output_dir=output_dir,  # type: ignore
+                    model=model, metadata=machine, output_dir=output_dir  # type: ignore
```

Comment:
Maybe a bit much for this PR, but if `machine` is now synonymous with `metadata`, perhaps renaming things accordingly would be good here, if not just the internal methods for now.

Reply:
ya.. I was on the fence about renaming as well and trying to minimize changes.. :man_shrugging:

Reply:
I'll make an issue for it.